src/Voron/Impl/Scratch/ScratchBufferFile.cs

```diff
@@ -48,9 +48,10 @@ public ScratchBufferFile(AbstractPager scratchPager, int scratchNumber)
             scratchPager.AllocatedInBytesFunc = () => AllocatedPagesCount * Constants.Storage.PageSize;
 
             _disposeOnceRunner = new DisposeOnce<SingleAttempt>(() =>
-            {
+            {                
                 _scratchPager.PagerState.DiscardOnTxCopy = true;
                 _scratchPager.Dispose();
+                Reset();
```

Comment:
What is this for here?

Reply:
Tens of millions of dictionary entries showed up in the original dump. The leak was probably from the _current of the scratch buffer pool but still clearing the dictionaries will release the memory faster.

Reply:
The intend is to clear the dictionaries, right? I'd rather prefer to call `.Clear()` on those dicts explicitly in `Dispose()` method or create method `ClearDictionaries()` and using from both: `Reset()` and `Dispose()` 

I don't expect any changes in `Reset()` method in the future but just in case let's try to avoid calling it during the disposal as it's supposed to reset the state of the scratch buffer file and continue using it, not to dispose / clear the resources.